docs/signing-spec.md

```diff
@@ -342,6 +343,236 @@ For example, if a package is signed by an unknown host key, then this package ha
+This would create a _strong yank_ under which the yank would invalidate all signatures.
+
+The main flaw with this route is that it makes a strong claim about what `yanked` means.
+In this case, `yanked` means "it is never secure to use this package", which may be true in some cases, but not all.
+
+Another issue with this approach is that a yanked bindle's non-host signatures would all be initially invalidated by the
+yank operation, even though the actual content of the package has not changed.
+This would require verifiers and creators to go _re-sign_ packages again after the `yanked` field was toggled to true.
+Thus, on yank, a bindle would become unusable until the signers re-signed.
+That was not the intent of the `yanked` field or the yanking operation.
+
+## Threat Model
+
+This section describes our threat model in applying signatures.
+Our inspiration has been [TUF]().
+However, Bindle's problem space is slightly different:
+- Bindle _is not_ concerned with updating a package. The identifier (name plus version) is "static" and must always point to the same artifact.
+- Bindle _is_ concerned with more than storage and retrieval.
+
+Likewise, Bindle is distinct from [in-toto]().
+While in-toto focuses on the assembly of the package (and how to attest and verify the steps of creating the package),
+Bindle's security model involves attesting and verifying the produced package at specific stages.
+It may be possible to map out Bindle's model using an in-toto static layout.
+
+Bindle's signing model is designed to provide a way for multiple entities (in different roles) to attach attestations to an immutable set of objects.
+Note that Bindle's security model is complementary to both TUF and in-toto: All three of these technologies could be used in conjunction.
+
+We assume that:
+
+- No single key holder can be assumed to be trustworthy
+    - A creator is generally an authority on what they wish to produce, but is not necessarily to be trusted to produce non-malicious bindles
+    - An approver, likewise, may produce results that are incomplete or potentially malicious
+    - A proxy or host may be malicious
+- An attacker can act as a man-in-the-middle (MITM) via hijacking or proxying. Therefore, the security model must assume a potential MITM.
+- If an attack can compromise enough keys, the best that can be done by this spec is to limit the distribution of compromised bindles
+- A malicious signer can cause a package's trust to be doubted (e.g. denial of service), but it should not be able to make a doubted package's trust _accepted_.
+    - That is, if other signatures are invalid, a malicious signature should not be able to override that invalidation
+    - But as a weakness in the system, if a malicious signer creates a bad signature, that signature may result in an uncompromised package becoming untrusted for any user who has the malicious signer's signature in their keyring
```

Comment:
This reads a little confusing to me. Are you saying that if a person has the key of the malicious signer in their local keyring it becomes untrusted? Maybe I'm not reading it right

Reply:
I will rephrase... but basically, you could perform a sort of denial of service attack on a package by signing it with a known key, but with a broken signature.

So, for example, an attacker could get ahold of Alice's public key, and if Alice had the ability to sign a package, they could create a broken signature with Alice's public key and completely garbage signature data.